diff/parse.go

```diff
@@ -223,8 +230,31 @@ func (r *FileDiffReader) HunksReader() *HunksReader {
 
 // ReadFileHeaders reads the unified file diff header (the lines that
 // start with "---" and "+++" with the orig/new file names and
-// timestamps).
+// timestamps). Or starts with "Only in " message with dir path filename.
 func (r *FileDiffReader) ReadFileHeaders() (origName, newName string, origTimestamp, newTimestamp *time.Time, err error) {
+	if (r.fileHeaderLine != nil) && (bytes.HasPrefix(r.fileHeaderLine, onlyInMessagePrefix)) {
+		path := bytes.Split(bytes.TrimPrefix(
+			bytes.TrimSuffix(r.fileHeaderLine, []byte("\n")),
+			onlyInMessagePrefix), []byte(":"))
+		if len(path) != 2 {
+			return "", "", nil, nil,
+				&ParseError{r.line, r.offset, ErrBadOnlyInMessage}
```

Comment:
Not sure if we want to throw an error here. Is it not allowed to have an extended header line that starts with "Only in ", but is not about an added/removed file?